src/StackInterface.php

```diff
@@ -43,5 +17,5 @@ public function withoutMiddleware(MiddlewareInterface $middleware);
      *
      * @return ResponseInterface
      */
-    public function process(RequestInterface $request);
+    public function dispatch(RequestInterface $request);
```

Comment:
> With the current interface a pipe/stack cannot be a middleware at the same time because the methods have the same name.

What's problematic, I agree.